src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/RazorCodeDocumentExtensions.cs

```diff
@@ -11,6 +11,8 @@ internal static class RazorCodeDocumentExtensions
     {
         private static readonly object UnsupportedKey = new object();
         private static readonly object SourceTextKey = new object();
+        private static readonly object CSharpSourceTextKey = new object();
```

Comment:
Weird that this should be a "key" when it's just an empty object. Makes sense, it's just not what I would have expected.

Reply:
Yup. This is what we use for storing arbitrary things in the CodeDocument. The other option is to use a string but using an object makes it so that no one else can retrieve it directly without using the methods defined here.